src/socket_base.cpp

```diff
@@ -1414,6 +1414,7 @@ void zmq::socket_base_t::update_pipe_options(int option_)
         for (pipes_t::size_type i = 0; i != pipes.size(); ++i)
         {
             pipes[i]->set_hwms(options.rcvhwm, options.sndhwm);
+            pipes[i]->set_peer_hwms(options.sndhwm, options.rcvhwm);
```

Comment:
@bluca  well ,you are right, I review the code, and found the peer pipe is also used by other thread.

And I find something. since the send_activate_write is called where `msgs_read % lwm == 0`
If we change hwm after connect, the behavior may be strange.

For example:
```
[Side A]
hwm = 150;
lwm = 75
msg_written = 100;

[Side B]
hwm = 150;
lwm = 75
msg_read = 50;
```
If we lower the sndhwm / rcvhwm, so that the hwm = 90, and lwm = 45
Then, we will get a full hwm (now the msg_written > 90), but actually we only has 50 pkg (50 unconfirmed);